nikola/plugins/task/archive.py

```diff
@@ -24,231 +24,146 @@
-
-                # Filter untranslated posts (via Issue #1360)
-                if not kw["show_untranslated_posts"]:
-                    posts = [p for p in posts if lang in p.translated_to]
-
-                if kw["create_monthly_archive"] or kw["create_full_archives"]:
-                    title = kw["messages"][lang]["Posts for {month} {year}"].format(
-                        year=year, month=nikola.utils.LocaleBorg().get_month_name(int(month), lang))
-                    yield self._generate_posts_task(kw, yearmonth, lang, posts, title)
-
-                if not kw["create_full_archives"] and not kw["create_daily_archive"]:
-                    continue  # Just to avoid nesting the other loop in this if
-                # Add archive per day
-                days = dict()
-                for p in posts:
-                    if p.date.day not in days:
-                        days[p.date.day] = list()
-                    days[p.date.day].append(p)
-                for day, posts in days.items():
-                    title = kw["messages"][lang]["Posts for {month} {day}, {year}"].format(
-                        year=year, month=nikola.utils.LocaleBorg().get_month_name(int(month), lang), day=day)
-                    yield self._generate_posts_task(kw, yearmonth + '/{0:02d}'.format(day), lang, posts, title)
-
-        if not kw['create_single_archive'] and not kw['create_full_archives']:
-            # And an "all your years" page for yearly and monthly archives
-            if "is_feed_stale" in kw:
-                del kw["is_feed_stale"]
-            years = list(self.site.posts_per_year.keys())
-            years.sort(reverse=True)
-            kw['years'] = years
-            for lang in kw["translations"]:
-                items = [(y, self.site.link("archive", y, lang), len(self.site.posts_per_year[y])) for y in years]
-                yield self._prepare_task(kw, None, lang, None, items, "list.tmpl", kw["messages"][lang]["Archive"])
-
-    def archive_path(self, name, lang, is_feed=False):
-        """Link to archive path, name is the year.
-
-        Example:
-
-        link://archive/2013 => /archives/2013/index.html
-        """
-        if is_feed:
-            extension = ".atom"
-            archive_file = os.path.splitext(self.site.config['ARCHIVE_FILENAME'])[0] + extension
-            index_file = os.path.splitext(self.site.config['INDEX_FILE'])[0] + extension
+        page_kind = "list"
+        if self.show_list_as_index:
+            if not self.show_list_as_subcategories_list or len(hierarchy) == self.max_levels:
+                page_kind = "index"
+        if len(hierarchy) == 0:
+            title = kw["messages"][lang]["Archive"]
+            kw["is_feed_stale"] = False
+        elif len(hierarchy) == 1:
+            title = kw["messages"][lang]["Posts for year %s"] % hierarchy[0]
+            kw["is_feed_stale"] = (datetime.datetime.utcnow().strftime("%Y") != hierarchy[0])
```

Comment:
I see this was in the old version, but is calling `utcnow` the correct behaviour? This can go wrong when someone is building when the thing is incorrect.

And what does `is_feed_stale` even mean? What if someone has backdated posts?